mysql-test/include/have_rocksdb_as_default.inc

```diff
@@ -0,0 +1,12 @@
+--source include/have_rocksdb.inc
```

Comment:
So testing this out and it doesn't work with the suite.opt as expected. suite.opt is `$ROCKSDB_OPT $ROCKSDB_LOAD_ADD --default-storage-engine=rocksdb --default-tmp-storage-engine=rocksdb`

and mtr incantation is `./mtr  --force --retry=0 --retry-failure=0 --max-test-fail=0 --no-warnings --testcase-timeout=600 --mysqld=--plugin-load="rocksdb=ha_rocksdb.so;rocksdb_cfstats=ha_rocksdb.so;rocksdb_dbstats=ha_rocksdb.so;rocksdb_perf_context=ha_rocksdb.so;rocksdb_perf_context_global=ha_rocksdb.so;rocksdb_cf_options=ha_rocksdb.so;rocksdb_global_info=ha_rocksdb.so;rocksdb_ddl=ha_rocksdb.so;rocksdb_index_file_map=ha_rocksdb.so" --suite=rocksdb`

Adding the `--mysqld=--default-storage-engine=rocksdb --mysqld=--default-tmp-storage-engine=rocksdb` back in works. The logs indicate it is the have_rocksdb_as_default.inc that is causing the skip. So somehow the --d-s-e and --d-t-s-e suite options are not getting passed into the server instance.

From a little poking and debugging, it looks like in  $default_storage_engine is set if specified on the command line, but it is not set if it is in suite.opt, this causes the logic in collect_one_test_case to take a different path and I believe the end result is that MyISAM becomes the default engine. So it seems that options in the suite.opts and test.opts are not run through the collect_options function which sets the global option variables. That makes sense because we wouldn't want a single test or suite option to change the global for other suites. So to fix this I think there would need to be a test by test check in the options for things like --d-s-e and perhaps others.

Honestly, at this point I'm not sure the depth of changes needed to fix this are worth the effort and risk of breaking mtr. Can you confirm my research here and advise if we should just leave as is and require --d-s-e and --d-t-s-e along with the --mysqld=--plugin-load="..." incantation in order to run rocksdb tests for now and perhaps revisit this at some later time if it becomes more critical? Perhaps we can convince upstream to fix their test cases to not pre-require rocksdb as the d-s-e which would solve it all for us anyway.